src/test/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageProcessorTest.java

```diff
@@ -0,0 +1,263 @@
+    private IMessageReceiver messageReceiver;
+    @Mock
+    private NotificationMessageHandler notificationMessageHandler;
+    @Mock
+    private NotificationMessageParser notificationMessageParser;
+
+    @BeforeEach
+    public void before() throws Exception {
+        notificationMessageProcessor = new NotificationMessageProcessor(
+            messageReceiver,
+            notificationMessageHandler,
+            notificationMessageParser,
+            5
+        );
+    }
+
+    @Test
+    public void should_return_true_when_there_is_a_message_to_process() throws Exception {
+        // given
+        IMessage message = mock(IMessage.class);
+        MessageBody messageBody = mock(MessageBody.class);
+        given(message.getMessageBody()).willReturn(messageBody);
+        given(messageReceiver.receive()).willReturn(message);
+
+        NotificationMsg notificationMsg = mock(NotificationMsg.class);
+        given(notificationMessageParser.parse(messageBody)).willReturn(notificationMsg);
+        doNothing().when(notificationMessageHandler).handleNotificationMessage(notificationMsg);
+
+        // when
+        boolean processedMessage = notificationMessageProcessor.processNextMessage();
+
+
+        // then
+        assertThat(processedMessage).isTrue();
+        verify(messageReceiver).receive();
+        verify(notificationMessageParser).parse(messageBody);
+        verify(notificationMessageHandler).handleNotificationMessage(notificationMsg);
+
+    }
+
+    @Test
+    public void should_return_false_when_there_is_no_message_to_process() throws Exception {
+        // given
+        given(messageReceiver.receive()).willReturn(null);
+
+        // when
+        boolean processedMessage = notificationMessageProcessor.processNextMessage();
+
+        // then
+        assertThat(processedMessage).isFalse();
+    }
+
+    @Test
+    public void should_not_throw_exception_when_queue_message_is_invalid() throws Exception {
+        IMessage message = mock(IMessage.class);
+        MessageBody messageBody = mock(MessageBody.class);
+        given(message.getMessageBody()).willReturn(messageBody);
+        given(messageReceiver.receive()).willReturn(message);
+        given(notificationMessageParser.parse(messageBody)).willThrow(new InvalidMessageException("Invalid Message"));
+
+        //when
+        assertThat(notificationMessageProcessor.processNextMessage()).isTrue();
+
+        // then
+        verify(messageReceiver).receive();
+        verify(notificationMessageParser).parse(messageBody);
+        verifyNoMoreInteractions(notificationMessageHandler);
+
+    }
+
+    @Test
+    public void should_not_throw_exception_when_notification_handler_fails() throws Exception {
+        // given
+        IMessage message = mock(IMessage.class);
+        MessageBody messageBody = mock(MessageBody.class);
+        given(message.getMessageBody()).willReturn(messageBody);
+
+        given(message.getMessageBody()).willReturn(messageBody);
```

Comment:
I guess this line is not needed, as it's the same as the previous one.

Reply:
removed